lib/internal/Magento/Framework/File/Csv.php

```diff
@@ -126,13 +126,17 @@ public function getDataPairs($file, $keyIndex = 0, $valueIndex = 1)
     /**
      * Saving data row array into file
      *
-     * @param   string $file
-     * @param   array $data
-     * @return  $this
+     * @param string $file
+     * @param array $data
+     * @param string $mode
+     *
+     * @return $this
+     *
+     * @throws \Magento\Framework\Exception\FileSystemException
      */
-    public function saveData($file, $data)
+    public function saveData($file, $data, $mode = 'w')
```

Comment:
Please, add new method parameters according our [Backward Compatible Development Guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#adding-parameters-in-public-methods)

Reply:
Done. But in my very humble opinion, it makes the code bloated. My first proposal was not breaking backward compatibility and kept the original code.